packages/api/src/router/notif.ts

```diff
@@ -42,6 +42,33 @@ export const notifRouter = createTRPCRouter({
       })
     }),
 
+  /**
+   *  @remarks
+   *  Adds current user to readByUserIds field in DB
+   *
+   *  @param  chatId - the id of the chat
+   *  @param  userId - the id of the logged in user
+   *  @param  type - the chat type (DIRECT or GROUP)
+   *  @returns an array of message objects
+   */
+  markChatAsReadByChatIdAndUserIdAndChatType: publicProcedure
```

Comment:
Is there a way to update a chat in a more generic way?

Reply:
I think its better to keep the apis seperated how they are for easier testing and implementation.